forbidigo/forbidigo.go

```diff
@@ -97,12 +106,29 @@ type visitor struct {
 	linter   *Linter
 	comments []*ast.CommentGroup
 
-	fset   *token.FileSet
-	issues []Issue
+	fset      *token.FileSet
+	typesInfo *types.Info
+	issues    []Issue
 }
 
-func (l *Linter) Run(fset *token.FileSet, nodes ...ast.Node) ([]Issue, error) {
-	var issues []Issue 
+func (l *Linter) RunForPackages(pkgs []*packages.Package) ([]Issue, error) {
+	var issues []Issue
+	for _, p := range pkgs {
+		nodes := make([]ast.Node, 0, len(p.Syntax))
+		for _, n := range p.Syntax {
+			nodes = append(nodes, n)
+		}
+		newIssues, err := l.Run(p.Fset, p.TypesInfo, nodes...)
+		if err != nil {
+			log.Fatalf("failed: %s", err)
```

Comment:
I'm still trying to determine if I like pushing the `RunForPackages` logic into this package, but at minimum, I think we should not be doing a fatal error in library code.  I recommend we just surface the error and let the caller decide if an error is fatal or not.

Reply:
True.

Reply:
Note that `textFor` already has a `log.Fatalf` in master.